### PR TITLE
Upgrade to storybook 8

### PR DESCRIPTION
**Pull request by georgeblahblah**
## What does this change?

- Updates DCR and AR to Storybook 8
- Updates the test for whether a component is being rendered in Storybook as the `__STORYBOOK_CLIENT_API__` property has been removed from `window`. This is replaced with `STORIES`
- Updates any incompatible configuration
- Applies a Webpack patch (thx @mxdvl!)

## What next?

- @arelra suggests to start fresh with Storybook as we've carried our config a long way now. This might be something for health sprint.

### CODE REVIEW

File: apps-rendering/package.json [hunk 1]
```diff
@@ -110,5 +110,8 @@
 		"whatwg-fetch": "3.6.19",
 		"winston": "3.11.0",
 		"winston-daily-rotate-file": "4.7.1"
+	},
+	"devDependencies": {
+		"@storybook/addon-webpack5-compiler-babel": "^3.0.3"
```
Comment (mxdvl) [inline apps-rendering/package.json:115]: Should this stay in the main `dependencies`?

File: dotcom-rendering/.storybook/main.ts [hunk 2]
```diff
@@ -107,11 +114,12 @@ const webpackConfig = (config: Configuration) => {
 
 	// modify storybook's file-loader rule to avoid conflicts with our svg
 	// https://stackoverflow.com/questions/54292667/react-storybook-svg-failed-to-execute-createelement-on-document
```
Comment (mxdvl) [inline dotcom-rendering/.storybook/main.ts:116]: Is this still relevant?

File: dotcom-rendering/src/components/FormatBoundary.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ export const FormatBoundary = ({ format, children }: Props) => {
 
 	const [isStorybook, setIsStorybook] = useState(false);
 	useEffect(() => {
-		if (!('__STORYBOOK_CLIENT_API__' in window)) return;
+		if (!('STORIES' in window)) return;
```
Comment (mxdvl) [inline dotcom-rendering/src/components/FormatBoundary.tsx:26]: Not for now, by maybe we ought to remove this entirely 🤔

**Review by mxdvl — APPROVED** [review]
8️⃣ 8️⃣ 8️⃣

File: apps-rendering/package.json (hunk 1 above)
Comment (georgeblahblah) [reply apps-rendering/package.json:115]: yes, good spot! storybook's auto-updater puts it here. I'll move it!

File: dotcom-rendering/.storybook/main.ts (hunk 2 above)
Comment (georgeblahblah) [reply dotcom-rendering/.storybook/main.ts:130]: the link to SO? I guess not! I'll remove it

**Pull request merged**